### PR TITLE
Feature/allow histogramming jagged variables

### PR DESCRIPTION
**Pull request by mafrahm**
This PR allows creating histograms with jagged variables, e.g.
```
cfg.add_variable(
    name="jets_pt",
    expression="Jet.pt",
    binning=(40, 0.0, 400.0),
    unit="GeV",
    x_title=r"$p_{T} of all jets$",
)
```
creates a single histogram that contains the pt of all jets.

### CODE REVIEW

File: columnflow/tasks/histograms.py [hunk 1]
```diff
@@ -199,9 +199,10 @@ def expr(events, *args, **kwargs):
                             return route.apply(events, null_value=variable_inst.null_value)
                     # apply it
                     fill_kwargs[variable_inst.name] = expr(events)
+
                 # broadcast and fill
```
Comment (riga) [inline columnflow/tasks/histograms.py:203]: Could you adjust the comment maybe? :)

File: columnflow/tasks/histograms.py [hunk 2]
```diff
@@ -185,7 +185,7 @@ def run(self):
                 fill_kwargs = {
                     "category": events.category_ids,
                     "process": events.process_id,
-                    "shift": self.global_shift_inst.id,
+                    "shift": ak.Array(np.zeros(len(events)) + self.global_shift_inst.id),
```
Comment (riga) [inline columnflow/tasks/histograms.py:188]: Is the `ak.Array` needed here? If not, then what is typically done is

```python
np.ones(len(events), dtype=np.int32) * self.global_shift_inst.id
```

Esp. the dtype could be needed here as `zeros` could default to `float64`.

**Review by riga — CHANGES_REQUESTED** [review]
Just two minor things inline and a question:

Would this also be required for histogramming cutflow variables in the other task chain?

File: columnflow/tasks/histograms.py (hunk 2 above)
Comment (mafrahm) [reply columnflow/tasks/histograms.py]: changed :+1:

**Comment by mafrahm** [comment]
> Would this also be required for histogramming cutflow variables in the other task chain?

Good point, it's implemented there aswell now

**Comment by pkausw** [comment]
@mafrahm can you fix the linting issue here? I think we could then proceed with approving this PR

**Comment by mafrahm** [comment]
> @mafrahm can you fix the linting issue here? I think we could then proceed with approving this PR

done!

File: columnflow/tasks/histograms.py (hunk 1 above)
Comment (riga) [reply columnflow/tasks/histograms.py:203]: Ok, after looking at the [docs](https://awkward-array.org/doc/main/reference/generated/ak.cartesian.html), "broadcast" still makes sense here.

**Review by riga — APPROVED** [review]

**Pull request merged**